src/ncp/ncp_spi.cpp

```diff
@@ -226,92 +231,69 @@ NcpSpi::SpiTransactionComplete(
     );
 }
 
-uint16_t
-NcpSpi::OutboundFrameSize(void)
+ThreadError NcpSpi::OutboundFrameBegin(void)
 {
-    return static_cast<uint16_t>(mSendFrameIter - (mSendFrame + SPI_HEADER_LENGTH));
+    return mTxFrameBuffer.InFrameBegin();
 }
 
-uint16_t
-NcpSpi::OutboundFrameGetRemaining(void)
+ThreadError NcpSpi::OutboundFrameFeedData(const uint8_t *aDataBuffer, uint16_t aDataBufferLength)
 {
-    return static_cast<uint16_t>((sizeof(mSendFrame) - SPI_HEADER_LENGTH) - OutboundFrameSize());
+    return mTxFrameBuffer.InFrameFeedData(aDataBuffer, aDataBufferLength);
 }
 
-ThreadError
-NcpSpi::OutboundFrameBegin(void)
+ThreadError NcpSpi::OutboundFrameFeedMessage(Message &aMessage)
 {
-    ThreadError errorCode( kThreadError_None );
-
-    if (mSending)
-    {
-        errorCode = kThreadError_Busy;
-    }
-    else
-    {
-        mSendFrameIter = (mSendFrame + SPI_HEADER_LENGTH);
-    }
+    return mTxFrameBuffer.InFrameFeedMessage(aMessage);
+}
 
-    return errorCode;
+ThreadError NcpSpi::OutboundFrameSend(void)
+{
+    return mTxFrameBuffer.InFrameEnd();
 }
 
-ThreadError
-NcpSpi::OutboundFrameFeedData(const uint8_t *frame, uint16_t frameLength)
+void NcpSpi::TxFrameBufferHasData(void *aContext, NcpFrameBuffer *aNcpFrameBuffer)
 {
-    ThreadError errorCode( kThreadError_None );
-    uint16_t maxOutLength( OutboundFrameGetRemaining() );
+    (void)aContext;
+    (void)aNcpFrameBuffer;
 
-    if (frameLength > maxOutLength)
-    {
-        errorCode = kThreadError_Failed;
-    }
-    else
-    {
-        memcpy(mSendFrameIter, frame, frameLength);
-        mSendFrameIter += frameLength;
-    }
+    sNcpSpi->TxFrameBufferHasData();
+}
 
-    return errorCode;
+void NcpSpi::TxFrameBufferHasData(void)
+{
+    PrepareNextSpiSendFrame();
 }
 
-ThreadError
-NcpSpi::OutboundFrameFeedMessage(Message &message)
+ThreadError NcpSpi::PrepareNextSpiSendFrame(void)
 {
-    ThreadError errorCode( kThreadError_None );
-    uint16_t maxOutLength( OutboundFrameGetRemaining() );
-    uint16_t frameLength( message.GetLength() );
+    ThreadError errorCode = kThreadError_None;
+    uint16_t    frameLength;
```

Comment:
To align the var declarations. I remove them (it's not commonly used in Ncp files)...